airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py

```diff
@@ -512,20 +514,25 @@ def request_params(
         stream_slice: Mapping[str, Any] = None,
         next_page_token: Mapping[str, Any] = None,
     ) -> MutableMapping[str, Any]:
-        next_page_token = next_page_token or {}
         parsed_state = self.check_stream_state(stream_state)
-        if self.cursor_field:
-            params = {"start_time": next_page_token.get(self.cursor_field, parsed_state)}
-        else:
-            params = {"start_time": calendar.timegm(pendulum.parse(self._start_date).utctimetuple())}
-        # check "start_time" is not in the future
-        params["start_time"] = self.check_start_time_param(params["start_time"])
+        params = {"start_time": self.check_start_time_param(parsed_state)}
         if self.sideload_param:
             params["include"] = self.sideload_param
         if next_page_token:
             params.update(next_page_token)
         return params
 
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
+        old_value = (current_stream_state or {}).get(self.cursor_field, 0)
+        new_value = (latest_record or {}).get(self.cursor_field, 0)
+        return {self.cursor_field: max(new_value, old_value)}
+
+    def check_stream_state(self, stream_state: Mapping[str, Any] = None):
```

Comment:
Please add the return type in the signature

Reply:
added